### DISCUSSION TITLE
Add advertising on a site loaded with github pages

### DISCUSSION THREAD

**Discussion by lucapu88** [community/community#22016] [Other Feature Feedback, Questions, & Ideas]
<p>Hello everyone, I have my small static site published with github pages, I would like to add advertising banners. Is it possible or would I be blocked / banned from github?</p>

**Comment by jfullstackdev** [comment]
<blockquote>
<h3>Prohibited uses</h3>
<p>GitHub Pages is not intended for or allowed to be used as a free web-hosting service to run your online business, e-commerce site, or any other website that is primarily directed at either facilitating commercial transactions or providing commercial software as a service (SaaS). GitHub Pages sites shouldn’t be used for sensitive transactions like sending passwords or credit card numbers.</p>
<p>In addition, your use of GitHub Pages is subject to the <a href="https://docs.github.com/en/free-pro-team@latest/github/site-policy/github-terms-of-service">GitHub Terms of Service</a>, including the restrictions on get-rich-quick schemes, sexually obscene content, and violent or threatening content or activity.</p>
</blockquote>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#limits-on-use-of-github-pages">
  <header class="source">
      <img src="https://docs.github.com/assets/cb-803/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#limits-on-use-of-github-pages" target="_blank" rel="noopener">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/47092464/181068260-b6fed838-c0fc-4563-ae35-2e8dc581eee2.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#limits-on-use-of-github-pages" target="_blank" rel="noopener">About GitHub Pages - GitHub Docs</a></h3>

  <p>You can use GitHub Pages to host a website about yourself, your organization, or your project directly from a repository on GitHub.com.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>you can add an advert but not a full commercial site, just like a donation button even in a GitHub repo, like external sponsors that are being posted</p>

**Comment by lucapu88** [comment]
<p>You may not have understood, my site does none of this, does not ask for passwords and no data. I asked if it is possible to add a pupup / banner ad, it is very different from asking for passwords, accounts or other things that involve transactions.</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="jdevstatic" data-post="2" data-topic="242395">
<div class="title">
<div class="quote-controls"></div>
 jdevstatic:</div>
<blockquote>
<p>you can add an advert but not a full commercial site</p>
</blockquote>
</aside>
<p>I doubt that, in my eyes advertisements are commercial activity even if it’s “just one” (assuming it’s for money, not just linking your friend’s blog or something). But only GitHub staff could give a definitive answer to that, the docs could be interpreted either way.</p>

**Comment by jfullstackdev** [comment, marked as answer]
<p>here is a not complete answer from a GitHub Staff</p>
<aside class="quote" data-post="2" data-topic="10660">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/can-i-set-google-adsense-advertisements-on-my-github-io-blog/10660/2">Can I set Google AdSense advertisements on my github.io blog?</a> <a class="badge-wrapper  bullet" href="/c/github-help/github-pages/31"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #c8e1ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Pages—websites for you and your projects hosted right on your GitHub repository.">GitHub Pages</span></a>
  </div>
  <blockquote>
    Hi <a class="mention" href="/zwl1619">@zwl1619</a>, 
Thank you for being here! I’m pretty sure Pages supports ad sense, thought to make that work you have to create a separate file then import it to your .yml.  Here is a <a href="https://flipdazed.github.io/blog/website/google-adsense-and-google-analytics" rel="nofollow noopener">blog post</a> with more specifics. 
I hope this helps!
  </blockquote>
</aside>

<p>my thinking is that, as long as it is not full commercial, just to make like a certain earning from donations, GitHub will not prevent that, that’s the same GitHub Sponsor, I’m seeing small ads also on a GitHub repo from the others and still it’s fine not banned,</p>
<p>I’m actually about to implement the same thing because the Philippines is not yet in the GitHub Sponsor list</p>

**Comment by harkerbyte** [comment]
Not a full commercial ads?

How about adsense on my popular repository... How sure it won't get blocked.. 

Or do we have any Google ads app from github marketplace

**Comment by gonsura** [comment]
can i also add shorten link adver (example : ouo.io) ?

**Comment by imtrinity94** [comment]
A nice example https://pscustomobject.github.io/
![image](https://user-images.githubusercontent.com/7029361/229704563-8aa53aa4-be4d-4d3c-ad52-35d64b5592a1.png)

**Reply by deepwilson** [reply]
@imtrinity94 Do you know how this user managed to have the base URL as https://pscustomobject.github.io/?

Usually, GitHub pages should have a repository name right? Hence, https://pscustomobject.github.io/ should give a 404 error.
However, this user has the repo name as the site name. Any idea how this is done?
Additionally, I also checked the _config.yaml but the url and baseurl is comented out? So how exactly is this working?
 
![image](https://github.com/community/community/assets/25292456/c7f4049e-145a-4b42-b3d1-c704f96e1b93)

**Reply by imtrinity94** [reply]
It's rather simple. For repos with same name as username, the URL is always username.github.io 
Check mine: 
https://imtrinity94.github.io/

**Reply by hh12github** [reply]
i changed my target practice website name to my username, but https://hh12github.github.io/hh12github/ is the name of the website now, any ideas on making https://hh12github.github.io not give a 404?

**Reply by shawshank725** [reply]
@hh12github The repo name should be hh12github.github.io. Then enable the github pages.

**Comment by deepwilson** [comment]
@imtrinity94 thank you. Was not aware of it!

**Comment by jimbethancourt** [comment]
[Advertising on Github](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies#10-advertising-on-github) indicates it is permissible, but "the primary focus of the Content posted in or through your Account to the Service should not be advertising or promotional marketing."
